azure-pipelines.yml

```diff
@@ -0,0 +1,105 @@
+# ASP.NET Core (.NET Framework)
+# Build and test ASP.NET Core projects targeting the full .NET Framework.
+# Add steps that publish symbols, save build artifacts, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/dotnet-core
+
+trigger:
+- master
+
+pool:
+  vmImage: 'macos-latest'
+
+variables:
+  buildConfiguration: 'Release'
+  mdocPath: 'bin/$(buildConfiguration)'
+  
+steps:
+- task: NuGetToolInstaller@1
+  displayName: Install NuGet Tool
+
+- task: Bash@3
+  displayName: Run Unit and Integration Tests
+  inputs:
+    targetType: 'inline'
+    script: 'make prepare all check CONFIGURATION=Debug'
+    
+- task: MSBuild@1
+  displayName: 'Building mdoc In $(buildConfiguration) Mode'
+  inputs:
+    solution: 'apidoctools.sln'
+    msbuildArguments: '/p:RunCodeAnalysis=False,RunCodeAnalysisOnThisProject=False'
+    configuration: '$(buildConfiguration)'
+
+- task: EsrpCodeSigning@1
+  displayName: Sign executable and dll files
+  condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
+  inputs:
+    ConnectedServiceName: 'CodeSigning-APEX'
+    FolderPath: '$(mdocPath)'
+    UseMinimatch: true
+    signConfigType: inlineSignParams
+    SessionTimeout: '60'
+    MaxConcurrency: '100'
+    MaxRetryAttempts: '5'
+    Pattern: |
+        *.dll
+        *.exe
+    inlineOperation: |
+      [
+        {
+          "KeyCode": "CP-236167",
+          "OperationSetCode": "SigntoolSign",
+          "parameters": [
+            {
+              "parameterName": "OpusName",
+              "parameterValue": "Microsoft"
+            },
+            {
+              "parameterName": "OpusInfo",
+              "parameterValue": "http://www.microsoft.com"
+            },
+            {
+              "parameterName": "PageHash",
+              "parameterValue": "/NPH"
+            },
+            {
+              "parameterName": "TimeStamp",
+              "parameterValue": "/tr \"http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer\" /td sha256"
+            },
+            {
+              "parameterName": "FileDigest",
+              "parameterValue": "/fd \"SHA256\""
+            }
+          ],
+          "ToolName": "sign",
+          "ToolVersion": "1.0"
+        },
+        {
+          "KeyCode": "CP-236167",
+          "OperationSetCode": "SigntoolVerify",
+          "Parameters": [
+            {
+              "parameterName": "VerifyAll",
+              "parameterValue": "/all"
+            }
+          ],
+          "ToolName": "sign",
+          "ToolVersion": "1.0"
+        }
+      ]
+
+- task: ArchiveFiles@2
+  displayName: Archive mdoc Files
+  inputs:
+    rootFolderOrFile: '$(mdocPath)'
+    includeRootFolder: false
+    archiveType: 'zip'
+    archiveFile: '$(Build.ArtifactStagingDirectory)/zips/mdoc-$(Build.BuildId).zip'
```

Comment:
for artifact version, shall we consider this format $(Year:yyyy).$(Month).$(DayOfMonth)$(Rev:.r) ?

Reply:
Yeah the artifact name I don't think matters that match, but adding the date into the name would definitely make it informative :)